### PR TITLE
Fix input field

### PR DESCRIPTION
**Pull request by nschubach**
## Description
Fixing an issue in the form input field and setting up for the confirm password component.

## Test Steps
1. Checkout this branch `git checkout fix-input-field`
2. Run fusion repo with linked blocks `npm run storybook`

### Before
Setting a custom error would cause the error state to get stuck in error.

### After
Setting a custom error is now properly cleared before checking validity.

## Author Checklist
- [x] Confirmed all the test steps a reviewer will follow above are working.
- [x] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [x] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [x] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [x] Confirmed this PR has unit test files
  - [x] Ran `npm run test`, made sure all tests are passing
  - [x] If the amount of work to write unit tests for this change are excessive,
please explain why (so that we can fix it whenever it gets refactored).
- [x] Confirmed relevant documentation has been updated/added.

## Reviewer Checklist
_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] Linting code actions have passed.
- [ ] Ran the code locally based on the test instructions.
  - [ ] I don’t think this is needed to be tested locally. For example, a padding style change (storybook?) or a logic change (write a test).
- [ ] I am a member of the engine theme team so that I can approve and merge this. If you're not on the team, you won't have access to approve and merge this pr.
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.

### CODE REVIEW

**Comment by JackHowa** [comment]
Screen Shot 2021-08-24 at 13 08 54
Screen Shot 2021-08-24 at 13 08 49
looks good

File: blocks/identity-block/components/FormInputField/index.jsx [hunk 1]
```diff
@@ -25,34 +25,35 @@ const FormInputField = ({
   validationPattern,
 }) => {
   const [valid, setValid] = useState(true);
+  const [value, setValue] = useState(defaultValue);
```
Comment (JackHowa) [inline blocks/identity-block/components/FormInputField/index.jsx:28]: wait, aren't we tracking state in the parent? this is new

Comment (nschubach) [reply blocks/identity-block/components/FormInputField/index.jsx:28]: yeah, I needed to hold that value to pass it along to the useEffect so that a change will trigger a revalidation.  I really didn't want to hold this value internally, but I don't see a way around it.

File: blocks/identity-block/components/FormInputField/index.jsx [hunk 2]
```diff
@@ -74,6 +75,7 @@ const FormInputField = ({
           ...(!valid ? ['xpmedia-form-field-input--error'] : []),
         ].join(' ')}
         id={inputId}
+        name={name}
```
Comment (JackHowa) [inline blocks/identity-block/components/FormInputField/index.jsx:80]: was this change deliberate? with field parameters

Comment (nschubach) [reply blocks/identity-block/components/FormInputField/index.jsx:80]: Yeah.  Name is required since it's used in the id for the field group so it HAS to be there.
```
  const infoId = `id_${name}_error`;
  const inputId = `id_${name}_input`;
```

**Review by JackHowa — APPROVED** [review]
looks good -- good find
